code/controllers/subsystem/economy.dm

```diff
@@ -78,21 +86,57 @@ SUBSYSTEM_DEF(economy)
 	bank_accounts_by_id = SSeconomy.bank_accounts_by_id
 	dep_cards = SSeconomy.dep_cards
 
+/// Processing step defines, to track what we've done so far
+#define ECON_DEPARTMENT_STEP "econ_dpt_stp"
+#define ECON_ACCOUNT_STEP "econ_act_stp"
+#define ECON_PRICE_UPDATE_STEP "econ_prc_stp"
+
 /datum/controller/subsystem/economy/fire(resumed = 0)
-	var/temporary_total = 0
 	var/delta_time = wait / (5 MINUTES)
-	departmental_payouts()
-	station_total = 0
-	station_target_buffer += STATION_TARGET_BUFFER
-	for(var/account in bank_accounts_by_id)
-		var/datum/bank_account/bank_account = bank_accounts_by_id[account]
-		if(bank_account?.account_job && !ispath(bank_account.account_job))
-			temporary_total += (bank_account.account_job.paycheck * STARTING_PAYCHECKS)
-		bank_account.payday(1)
-		station_total += bank_account.account_balance
-	station_target = max(round(temporary_total / max(bank_accounts_by_id.len * 2, 1)) + station_target_buffer, 1)
-	if(!HAS_TRAIT(SSeconomy, TRAIT_MARKET_CRASHING))
-		price_update()
+
+	if(!resumed)
+		temporary_total = 0
+		processing_part = ECON_DEPARTMENT_STEP
+		cached_processing = department_accounts.Copy()
```

Comment:
shouldnt this be bank_accounts_by_id

Reply:
no, because it's for `departmental_payouts()`, which was formerly done unlike the standard station target stuff as a subproc